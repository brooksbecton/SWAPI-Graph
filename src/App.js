import React, { Component } from "react";
import vis from "vis";

import SwapiGraphInterface from "./lib/SwapiGraphInterface";

class App extends Component {
  constructor() {
    super();

    this.state = {
      nodes: []
    };
  }

  componentDidMount = async () => {
    this.getNodes();
  };

  drawGraph = () => {
    // create an array with nodes
    var nodes = new vis.DataSet(this.state.nodes);

    // create an array with edges
    var edges = new vis.DataSet([]);

    // create a network
    var container = document.getElementById("mynetwork");

    // provide the data in the vis format
    var data = {
      nodes: nodes,
      edges: edges
    };
    var options = {
      physics: {
        enabled: false
      }
    };

    // initialize your network!
    var network = new vis.Network(container, data, options);
  };

  getNodes = async () => {
    //TODO hit the root API https://swapi.co/api and get what collections they have for less work later.
    const peopleInterface = SwapiGraphInterface("https://swapi.co/api/people/");
    const peopleNodes = await peopleInterface.getCollectionItems();

    const filmsInterface = SwapiGraphInterface("https://swapi.co/api/films/");
    const filmsNodes = await filmsInterface.getCollectionItems();

    const planetsInterface = SwapiGraphInterface(
      "https://swapi.co/api/planets/"
    );
    const planetsNodes = await planetsInterface.getCollectionItems();

    const speciesInterface = SwapiGraphInterface(
      "https://swapi.co/api/species/"
    );
    const speciesNodes = await speciesInterface.getCollectionItems();

    const starshipsInterface = SwapiGraphInterface(
      "https://swapi.co/api/starships/"
    );
    const starshipsNodes = await starshipsInterface.getCollectionItems();

    const vehiclesInterface = SwapiGraphInterface(
      "https://swapi.co/api/vehicles/"
    );
    const vehiclesNodes = await vehiclesInterface.getCollectionItems();

    this.setState({
      nodes: this.state.nodes.concat(
        peopleNodes,
        filmsNodes,
        planetsNodes,
        speciesNodes,
        starshipsNodes,
        vehiclesNodes
      )
    });

    this.drawGraph();
  };

  render() {
    return <div id="mynetwork" />;
  }
}

export default App;
