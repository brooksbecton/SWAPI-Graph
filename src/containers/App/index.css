@import "~antd/dist/antd.css";

.content {
  background-color: white;
  margin: auto;
  margin: 0 50px;
  margin-top: 60px;
  min-height: 90vh;
  padding: 24px;
}

.topNav {
  background: transparent;
  position: fixed;
  width: 100%;
}
