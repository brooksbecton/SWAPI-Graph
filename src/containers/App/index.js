import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Layout } from "antd";

import "./index.css";

import About from "./../About";
import Graph from "./../Graph";
import Home from "./../Home";

import TopNav from "./../../components/TopNav";

const { Header, Content } = Layout;

const App = () => {
  return (
    <Router>
      <Layout>
        <Header className="topNav">
          <TopNav />
        </Header>
        <Content className="content">
          <Route exact path="/about" component={About} />
          <Route exact path="/home" component={Home} />
          <Route exact path="/" component={Home} />
          <Route exact path="/graph" component={Graph} />
        </Content>
      </Layout>
    </Router>
  );
};

export default App;
