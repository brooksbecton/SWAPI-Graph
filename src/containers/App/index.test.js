import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import App from "./index";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("renders", () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
