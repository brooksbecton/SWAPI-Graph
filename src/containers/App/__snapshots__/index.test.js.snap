// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "basename": "",
      "children": <Adapter>
        <OfflineChecker />
        <Adapter
                className="topNav"
        >
                <withRouter(TopNav) />
        </Adapter>
        <Adapter
                className="content"
        >
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/about"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/home"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/graph"
                />
        </Adapter>
</Adapter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <OfflineChecker />,
          <Adapter
            className="topNav"
>
            <withRouter(TopNav) />
</Adapter>,
          <Adapter
            className="content"
>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/about"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/home"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/graph"
            />
</Adapter>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(TopNav) />,
            "className": "topNav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/about"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/home"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/graph"
/>,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/graph",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "basename": "",
        "children": <Adapter>
          <OfflineChecker />
          <Adapter
                    className="topNav"
          >
                    <withRouter(TopNav) />
          </Adapter>
          <Adapter
                    className="content"
          >
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/about"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/home"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/graph"
                    />
          </Adapter>
</Adapter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <OfflineChecker />,
            <Adapter
              className="topNav"
>
              <withRouter(TopNav) />
</Adapter>,
            <Adapter
              className="content"
>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/about"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/home"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/graph"
              />
</Adapter>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withRouter(TopNav) />,
              "className": "topNav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/about"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/home"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/graph"
/>,
              ],
              "className": "content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/about",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/home",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/graph",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
